vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -141,14 +129,65 @@ def _transformed_inputs(self) -> Union[list[State], dict[str, Union[State, Contr
         # have already been classified as legacy in Action._legacy. In future when vm.Action.inputs is deprecated
         # then this will be used for vm.Action instances also.
         TypeAdapter(dict[str, _DotSeparatedStr]).validate_python(self._runtime_args)
-
+        # Do exactly same lookup but we only have "__default__" defined in inputs dictionary so will only ever use that
+        # case
         # User specified arguments runtime_args take precedence over built in reserved arguments. No static arguments
         # ar relevant here, just Dash States. Static arguments values are stored in the state of the relevant
         # _AbstractAction instance.
-        runtime_args = {arg_name: State(*arg_value.split(".")) for arg_name, arg_value in self._runtime_args.items()}
+        # Qn: should this work for legacy actions too? Think about docs.
+        runtime_args = {
+            arg_name: State(*self._transform(arg_value, type="input").split("."))
+            for arg_name, arg_value in self._runtime_args.items()
+        }
 
         return builtin_args | runtime_args
 
+    @staticmethod
+    def _transform(reference: str, type: Literal["output", "input"]) -> str:
```

Comment:
```suggestion
    def _transform_dependency(dependency: str, type: Literal["output", "input"]) -> str:
```

Input/Output/State all subclass `DashDependency`. I don't much like the name `dependency` but I think it's a bit clearer than `reference`.